tasks/main-tang.yml

```diff
@@ -1,81 +1,51 @@
 ---
 - name: Ensure tang is installed
   package:
-    name: "{{ __nbde_server_package_list }}"
+    name: "{{ __nbde_server_packages }}"
     state: present
 
 
-- name: Ensure tang is enabled and started
-  when: state is not defined or state != "stopped"
-  service:
-    name: "{{ __nbde_server_service }}"
-    enabled: yes
-    state: started
-
-
-- name: Ensure tang is enabled but stopped
-  when: state is defined and state == "stopped"
-  service:
-    name: "{{ __nbde_server_service }}"
-    enabled: yes
-    state: stopped
-
-
 - name: Ensure keys are rotated
   when: nbde_server_rotate_keys
   nbde_server_tang:
-    state: keys-rotated
+    action: rotate-keys
     keygen: "{{ __nbde_server_keygen }}"
     keydir: "{{ __nbde_server_keydir }}"
+  notify: nbde_server tang cache update
 
 
 - name: Ensure we have keys
   when: not nbde_server_rotate_keys
   nbde_server_tang:
-    state: keys-created
+    action: create-keys
     keygen: "{{ __nbde_server_keygen }}"
     keydir: "{{ __nbde_server_keydir }}"
+  notify: nbde_server tang cache update
 
 
-- name: Ensure we fetch the server keys
-  when: nbde_server_fetch_keys
-  synchronize:
-    mode: pull
-    src: "{{ __nbde_server_keydir }}/"
-    dest: "{{ nbde_server_keys_dir }}/{{ inventory_hostname }}"
-    delete: yes
-    rsync_opts:
-      - "--chmod=D2755,F644"
+- name: Perform key management (fetch/deploy) tasks
+  when: nbde_server_fetch_keys or nbde_server_deploy_keys
+  include: tang-key-management.yml
+  notify: nbde_server tang cache update
 
 
-- name: Ensure we fetch the created keys for deploying
-  when: nbde_server_deploy_keys
-  run_once: yes
-  synchronize:
-    mode: pull
-    src: "{{ __nbde_server_keydir }}/"
-    dest: "{{ nbde_server_keys_dir }}"
-    delete: yes
-    rsync_opts:
-      - "--chmod=D2755,F644"
-
-
-- name: Ensure we copy specific keys to tang
-  when: nbde_server_deploy_keys
-  check_mode: no
-  synchronize:
-    mode: push
-    src: "{{ nbde_server_keys_dir }}"
-    dest: "{{ __nbde_server_keys_to_deploy_dir }}"
+- name: Ensure tang is enabled and started
+  when: state is not defined or state != "stopped"
+  service:
+    name: "{{ __nbde_server_service }}"
+    enabled: yes
+    state: started
+  notify: nbde_server tang cache update
```

Comment:
I don't understand what this is doing?

Reply:
This should be enabling and starting the tang service.

Reply:
Where is `state` set?  Is that a variable that should be set by the user?

I guess I'm not understanding why you need a `when` in a `service`

Is the workflow not like this: 1) stop tang service 2) do some sort of update 3) start tang ?  You can't do 1) some sort of update 2) restart tang?

Reply:
Yeah, this looks wrong. The requirement is to have the service started by default, but have the ability to stop it. I guess we can rename this variable to `nbde_server_status`, have it as `started` by default, and have a single task to take care of the status. And document it properly. What do you think?

Reply:
And I meant `nbde_server_state` (typo). Perhaps `nbde_server_service_state`, instead.

Reply:
> The requirement is to have the service started by default, but have the ability to stop it.

There is a requirement that the tang service is not running after performing an install?

Reply:
The requirement is to be able to stop it, not necessarily after the install.  So, by default, it would be started, but at some point one may want to stop it for some maintenance/backup/restore of keys.

Reply:
> The requirement is to be able to stop it, not necessarily after the install. So, by default, it would be started, but at some point one may want to stop it for some maintenance/backup/restore of keys.

I replied to the wrong comment - see https://github.com/linux-system-roles/nbde_server/pull/6#discussion_r437022471

Reply:
Thanks for the insights. I am now enabling/starting the services, to follow the pattern in other roles.